tools/src/main/scala/caliban/tools/RemoteSchema.scala

```diff
@@ -0,0 +1,313 @@
+      locations = definition.locations.map(toDirectiveLocation(_)).toSet
+    )
+
+  private def toDirectiveLocation(loc: DirectiveLocation): __DirectiveLocation =
+    loc match {
+      case DirectiveLocation.ExecutableDirectiveLocation.QUERY                  => __DirectiveLocation.QUERY
+      case DirectiveLocation.ExecutableDirectiveLocation.MUTATION               => __DirectiveLocation.MUTATION
+      case DirectiveLocation.ExecutableDirectiveLocation.SUBSCRIPTION           => __DirectiveLocation.SUBSCRIPTION
+      case DirectiveLocation.ExecutableDirectiveLocation.FIELD                  => __DirectiveLocation.FIELD
+      case DirectiveLocation.ExecutableDirectiveLocation.FRAGMENT_DEFINITION    => __DirectiveLocation.FRAGMENT_DEFINITION
+      case DirectiveLocation.ExecutableDirectiveLocation.FRAGMENT_SPREAD        => __DirectiveLocation.FRAGMENT_SPREAD
+      case DirectiveLocation.ExecutableDirectiveLocation.INLINE_FRAGMENT        => __DirectiveLocation.INLINE_FRAGMENT
+      case DirectiveLocation.TypeSystemDirectiveLocation.SCHEMA                 => __DirectiveLocation.SCHEMA
+      case DirectiveLocation.TypeSystemDirectiveLocation.SCALAR                 => __DirectiveLocation.SCALAR
+      case DirectiveLocation.TypeSystemDirectiveLocation.OBJECT                 => __DirectiveLocation.OBJECT
+      case DirectiveLocation.TypeSystemDirectiveLocation.FIELD_DEFINITION       => __DirectiveLocation.FIELD_DEFINITION
+      case DirectiveLocation.TypeSystemDirectiveLocation.ARGUMENT_DEFINITION    => __DirectiveLocation.ARGUMENT_DEFINITION
+      case DirectiveLocation.TypeSystemDirectiveLocation.INTERFACE              => __DirectiveLocation.INTERFACE
+      case DirectiveLocation.TypeSystemDirectiveLocation.UNION                  => __DirectiveLocation.UNION
+      case DirectiveLocation.TypeSystemDirectiveLocation.ENUM                   => __DirectiveLocation.ENUM
+      case DirectiveLocation.TypeSystemDirectiveLocation.ENUM_VALUE             => __DirectiveLocation.ENUM_VALUE
+      case DirectiveLocation.TypeSystemDirectiveLocation.INPUT_OBJECT           => __DirectiveLocation.INPUT_OBJECT
+      case DirectiveLocation.TypeSystemDirectiveLocation.INPUT_FIELD_DEFINITION =>
+        __DirectiveLocation.INPUT_FIELD_DEFINITION
+    }
+
+  private def filterDeprecated(x: __Field, deprecated: __DeprecatedArgs): Boolean =
+    if (deprecated.includeDeprecated.getOrElse(true)) true
+    else !x.isDeprecated
+
+  private def filterDeprecated(x: __EnumValue, deprecated: __DeprecatedArgs): Boolean =
+    if (deprecated.includeDeprecated.getOrElse(true)) true
+    else !x.isDeprecated
+
+  private def isDeprecated(directives: List[Directive]): Boolean =
+    deprecationReason(directives).isDefined
+
+  private def deprecationReason(directives: List[Directive]): Option[String] =
+    directives.collectFirst {
+      case d if (d.name == "deprecated") =>
+        d.arguments
+          .get("reason")
+          .collect({ case StringValue(value) =>
```

Comment:
Nit: Don't need the parens around the `{`

Reply:
👍